packages/conversation/src/rest/v1/app/app-api.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  RequestBody,
+  SinchClientParameters,
+} from '@sinch/sdk-client';
+import {
+  AppCreateRequest,
+  AppResponse,
+  AppUpdateRequest,
+  ListAppsResponse,
+} from '../../../models';
+import { ConversationDomainApi } from '../conversation-domain-api';
+
+export interface CreateAppRequestData {
+  /** The app to create. */
+  'appCreateRequestBody': AppCreateRequest;
+}
+export interface DeleteAppRequestData {
+  /** The unique ID of the app. You can find this on the [Sinch Dashboard](https://dashboard.sinch.com/convapi/apps). */
+  'app_id': string;
+}
+export interface GetAppRequestData {
+  /** The unique ID of the app. You can find this on the [Sinch Dashboard](https://dashboard.sinch.com/convapi/apps). */
+  'app_id': string;
+}
+export interface ListAppsRequestData {
+}
+export interface UpdateAppRequestData {
+  /** The unique ID of the app. You can find this on the [Sinch Dashboard](https://dashboard.sinch.com/convapi/apps). */
+  'app_id': string;
+  /** The updated app. */
+  'appUpdateRequestBody': AppUpdateRequest;
+  /** The set of field mask paths. */
+  'update_mask'?: Array<string>;
```

Comment:
What about simplifying SDK usage but computing `update_mask` from provided `appUpdateRequestBody` content ?
e.g.
- appUpdateRequestBody do not contains a `display_name` value (i.e. `undefined` or key is not present): the update_mask will not contains `display_name` keyword to not handle the value from payload
- appUpdateRequestBody  contains a `display_name` value (i.e. `null` or `any string`): the update_mask will contains `display_name` keyword to set the value according to payload

Then the `update_mask`is no longer required on function call and it simplify call and logic onto SDK usage
This could improve DX (see https://github.com/sinch/sinch-sdk-node/pull/15/files#diff-5f095c61b28d52390193b18e6684b856802e801978662161d34257e25e9d0b26R18) and providing added value from SDKs above pure REST calls

Reply:
I don't concur with the analysis above: the user may want to send some data in the body (for instance copied from a previous `GET` request from the contact or another contact) and update only a subpart of it: this way, by specifying the property in the `update_mask` all the other properties will be ignored.
Suppressing the parameter in the public api and/or filling it automatically may improve DX but it degrades the API flexibility. Hence I'm not in favour of removing it from the request data interface

Reply:
What about adding override function, one with auto `update_mask`
```
Update(Contact)
```
And second for distinguished users which will override automatically set:
```
Update(Contact, UpdateMask)
```

Reply:
> I don't concur with the analysis above: the user may want to send some data in the body (for instance copied from a previous `GET` request from the contact or another contact) and update only a subpart of it: this way, by specifying the property in the `update_mask` all the other properties will be ignored. Suppressing the parameter in the public api and/or filling it automatically may improve DX but it degrades the API flexibility. Hence I'm not in favour of removing it from the request data interface

To add: if you won't touch other properties in this case they will be the same as before anyway so you will still achieve desired result

Reply:
```
var contact = Get() { 'display_name': 'a', 'channel': 'b' }
contact.channel = 'c'
Update(contact) { 'display_name': 'a', 'channel': 'c' }
```
The the same result above will be when you set `update_mask` to `channel`.

Reply:
I'm sorry but I still don't understand your point: in which use case do you want to force a user to use an parameter defined as optional by the API?

Reply:
ah, right, it's optional, then doesn't matter